### PR TITLE
[#311] Use stdint.h and stdbool.h for communication protocol

### PR DESCRIPTION
**Pull request by shikharish**

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Can you reenable the MacOS build in https://github.com/pgmoneta/pgmoneta/blob/main/.github/workflows/ci.yml and see how it goes ?

File: src/include/info.h [hunk 1]
```diff
@@ -67,14 +67,14 @@ struct backup
 {
    char label[MISC_LENGTH];                                       /**< The label of the backup */
    char wal[MISC_LENGTH];                                         /**< The name of the WAL file */
-   unsigned long backup_size;                                     /**< The backup size */
-   unsigned long restore_size;                                    /**< The restore size */
-   int elapsed_time;                                              /**< The elapsed time in seconds */
-   int version;                                                   /**< The version */
-   int minor_version;                                             /**< The minor version */
+   uint64_t backup_size;                                     /**< The backup size */
```
Comment (jesperpedersen) [inline src/include/info.h:70]: Align comments

File: src/libpgmoneta/json.c [hunk 2]
```diff
@@ -40,6 +41,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <stdint.h>
```
Comment (jesperpedersen) [inline src/libpgmoneta/json.c:44]: Sorting

File: src/libpgmoneta/json.c [hunk 3]
```diff
@@ -613,6 +615,96 @@ pgmoneta_json_get_value(struct json* item, char* key)
...
 }
 
+int8_t
+pgmoneta_json_get_int8_value(struct json* item, char* key)
+{
+   if (item == NULL)
+   {
+      return 0;
+   }
+   struct json_value* val = pgmoneta_json_get_value(item, key);
```
Comment (jesperpedersen) [inline src/libpgmoneta/json.c:625]: Can you move `struct json_value* val = NULL;` to the beginning of the function ?

File: src/libpgmoneta/management.c [hunk 4]
```diff
@@ -2808,47 +2901,54 @@ read_list_backup_json(SSL* ssl, int socket, char* server)
       {
          goto error;
       }
-      pgmoneta_json_item_put_int64(status, "Number of backups", number_of_backups);
+      pgmoneta_json_item_put_int32(status, "Number of backups", number_of_backups);
+      // number_of_backups=1;
```
Comment (jesperpedersen) [inline src/libpgmoneta/management.c:2905]: Remove

File: src/libpgmoneta/management.c [hunk 5]
```diff
@@ -2808,47 +2901,54 @@ read_list_backup_json(SSL* ssl, int socket, char* server)
...
       for (int i = 0; i < number_of_backups; i++)
       {
+         pgmoneta_log_info("n_backups %d", number_of_backups);
```
Comment (jesperpedersen) [inline src/libpgmoneta/management.c:2908]: Remove

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
There is something wrong...

Look at the output from the `info` command; `-F json` works, but `-F text` doesn't

**Comment by shikharish** [comment]
@jesperpedersen Should work now.

**Pull request merged**